### PR TITLE
New formula for clang-omp

### PR DESCRIPTION
**Pull request by totalvoidness**
Builds clang-omp (http://clang-omp.github.io), a clang with support for OpenMP pragmas.

### CODE REVIEW

**Comment by tdsmith** [comment]
libiomp was added in #33118; does this need additional work?

**Comment by totalvoidness** [comment]
I still don't know, why it didn't build on the CI server before. libiomp is not yet a dependency in the formula. Still I can of course rewrite the test to compile something, which actually uses libiomp and not just omp #pragmas.

**Comment by tdsmith** [comment]
@BrewTestBot test this please

**Comment by tdsmith** [comment]
Hmm, I'm almost ready to pull this but the version scheme is lousy. I've asked upstream for clarification about version tags.

**Comment by totalvoidness** [comment]
What happened to the build? Out of memory on yosemite, but success on the VMs?

File: Library/Formula/clang-omp.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  needs :cxx11
+
+  def install
+    resource("compiler-rt").stage { (buildpath/"projects/compiler-rt").install Dir["*"] }
+    resource("clang").stage { (buildpath/"tools/clang").install Dir["*"] }
+
+    system "./configure", "--prefix=#{prefix}", "--enable-cxx11", "--enable-libcpp"
+    system "make", "install"
+
+    mv bin/"clang", bin/"clang-omp"
```
Comment (mikemcquaid) [inline Library/Formula/clang-omp.rb]: Does this install anything else that shadows stuff in `/usr/bin`?

Comment (totalvoidness) [inline Library/Formula/clang-omp.rb]: Yes there are a few others. However this particular binary would interfere with projects, that rely on the "normal" clang. 

Is there a good way to install _only_ the n relevant out of m produced binaries to /usr/local/bin?

Comment (mikemcquaid) [inline Library/Formula/clang-omp.rb]: Could make `libexec` the prefix and write wrapper scripts/make symlinks into `bin`. That's probably the best call with stuff like this that we want isolated from the system a bit.

Comment (totalvoidness) [inline Library/Formula/clang-omp.rb]: Thanks, will give it a try. However there is still the problem of nondeterministic behaviour of Jenkins (see latest log file for Yosemite).

Comment (mikemcquaid) [inline Library/Formula/clang-omp.rb]: I've just restarted the build. That's not a Jenkins bug, Jenkins isn't doing anything special here.

Comment (totalvoidness) [inline Library/Formula/clang-omp.rb]: It's not Jenkins itself, but there are numerous threads about the following two lines, that lead to too limited memory:

    clang: error: unable to execute command: Killed: 9
    clang: error: clang frontend command failed due to signal (use -v to see invocation)

The Build has finished successfully on Yosemite before. Also on my local machine (same Yosemite release, same Xcode version) everything is fine.

Comment (mikemcquaid) [inline Library/Formula/clang-omp.rb]: The box it's running on has 16GB of memory so it seems unlikely that it is that.

Comment (totalvoidness) [inline Library/Formula/clang-omp.rb]: Strange... :disappointed:

**Comment by totalvoidness** [comment]
@tdsmith Can this be merged now?

**Comment by tdsmith** [comment]
Yep! Thanks very much for the PR and for your patience.

**Comment by mikemcquaid** [comment]
@tdsmith Looks like the bottles for this and `tor` didn't publish for some reason. Make sure to check with `brew fetch` before pushing, thanks.

**Comment by tdsmith** [comment]
I think the emails are spurious or there's some issues in Bintray's CDN; I successfully fetched before I pushed and the files still looked fine right after the test bot emails landed. Did you see something else?

**Comment by mikemcquaid** [comment]
Yeh, weird. Fails for me and the bot. CDN issues I think.

**Comment by tdsmith** [comment]
Bintray confirms: https://twitter.com/kumokasumi/status/585523841067278336

**Comment by DomT4** [comment]
Their fallback seems to be working. I can grab the ` tor ` bottle with ` fetch ` without issue now.

**Pull request closed without merging**